scripts/xtensa-build-zephyr.py

```diff
@@ -597,13 +599,17 @@ def clean_staging(platform):
 # detailed comments in west.yml
 RIMAGE_SOURCE_DIR = west_top / "sof" / "tools" / "rimage"
 
+platform_wconfig = None
+
 
 def rimage_west_configuration(platform_dict, dest_dir):
 	"""Configure rimage in a new file `dest_dir/westconfig.ini`, starting
 	from the workspace .west/config.
 	Returns the pathlib.Path to the new file.
 	"""
 
+	global platform_wconfig
```

Comment:
@marc-hb not sure why this is bad. The contents of the variable is platform-specific, but the variable itself is generic, why is it particularly bad to make it global?

Reply:
Because non-constant, global variables are bad generally speaking? This breaks referential transparency, now there is a hidden dependency between `rimage_west_configuration()` and `install_lib()`. Also, the build cannot be parallel anymore.

We had bugs exactly like that: 6bedd8e7427c654a